### PR TITLE
Add ability to create modules that contain additional metadata 

### PR DESCRIPTION
**Pull request by gverner**
Add ability to create Resource Adapters modules that contain additional metadata including META-INF/ra.xml, MANIFEST.MF, license files etc.

The metadata would be stored a folder located in the project /src/main/resources defined in using the same name that is used to define the module in build.grade. During the makeModules task these files are copied into the module folder, module.xml and jar files are excluded.

Modification to copy the metadata can be found in the makeLocal method of JbossModule and an additional property metadataDir, with default of 'resources' has been added to JBossServer, this allows the location to be modified if necessary.

Added an ActiveMQ Resource Adapter Module to the Sample Project to show how to use this feature.

Modified the JBossModule1_(1-6)Tests to create META-INF/ra.xml file in the resources.slf4j folder and verify that is copied to the module. 

Modified the JBossModule1_(1-6)Tests to create module.xml file in the resources.slf4j folder and verify that is is NOT copied to the module. 

Changed the version to 0.19, not sure if this was the right thing to do.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Java class files
```
Comment (zhurlik) [inline .gitignore]: That's good idea for adding .gitignore. Would be great to have a look at https://github.com/github/gitignore where we can use at least the following:
- https://github.com/github/gitignore/blob/master/Global/JetBrains.gitignore
- https://github.com/github/gitignore/blob/master/Java.gitignore

Comment (gverner) [reply .gitignore]: replace .gitignore with the contents of the following
https://github.com/github/gitignore/blob/master/Java.gitignore
https://github.com/github/gitignore/blob/master/Global/JetBrains.gitignore
https://github.com/github/gitignore/blob/master/Gradle.gitignore

Plus these 
# this gradle-jboss-modules project
tmp-repo/
# ignore sub-project gradle build
build/

With this config, picked up 4 Intellij files

File: src/main/groovy/com/github/zhurlik/extension/JBossModule.groovy [hunk 2]
```diff
@@ -208,7 +208,24 @@ class JBossModule {
                 }
             }
 
+            // copy static files from src/main/<metadataDir>/<moduleName> into module
```
Comment (zhurlik) [inline src/main/groovy/com/github/zhurlik/extension/JBossModule.groovy]: I think that according to https://docs.gradle.org/current/userguide/distribution_plugin.html#sec:distribution_contents
we can use `src/$distribution.name/dist` it means that we don't need to have `src/main/metadata dir`. 

All files under `src/${jboss-server}/dist/modules/${path-to-module}/test.txt` will be copied by gradle distribution plugin.

You can see how it works here:
1. module definition
https://github.com/zhurlik/gradle-jboss-modules-plugin/blob/master/samples/build.gradle#L246-L264
2. custom files for module
https://github.com/zhurlik/gradle-jboss-modules-plugin/tree/master/samples/src/wildfly8.2/dist/modules/com/github/zhurlik/b

I guess that you can just keep your changes in the https://github.com/gverner/gradle-jboss-modules-plugin/tree/master/samples/src but move files required for RA to `src/wildfly8.2/dist/modules/org/apache/activemq`

After getting your expected result we can just update the sample and the read.me to see how it works.
What do you think about this?

Comment (gverner) [reply src/main/groovy/com/github/zhurlik/extension/JBossModule.groovy]: Yes this works fine for me, I will back out my changes leaving just the Resource Adapter Sample.

**Comment by zhurlik** [comment]
Let me know when PR will be ready

**Comment by gverner** [comment]
PR is complete unless you have more suggestions.

**Pull request merged**